src/USER-MDI/fix_mdi_engine.cpp

```diff
@@ -228,9 +228,11 @@ int FixMDIEngine::execute_command(const char *command, MDI_Comm mdicomm)
   // respond to any driver command
 
   if (strcmp(command, ">NATOMS") == 0) {
-    ierr = MDI_Recv((char *) &atom->natoms, 1, MDI_INT, mdicomm);
+    int mdi_natoms = 0;
+    ierr = MDI_Recv((char *) &mdi_natoms, 1, MDI_INT, mdicomm);
     if (ierr != 0) error->all(FLERR, "MDI: Unable to receive number of atoms from driver");
-    MPI_Bcast(&atom->natoms, 1, MPI_INT, 0, world);
+    atom->natoms = mdi_natoms;
+    MPI_Bcast(&atom->natoms, 1, MPI_LMP_BIGINT, 0, world);
```

Comment:
@taylor-a-barnes @sjplimp this change is simply to correct the MPI_Bcast().
But should we simply overwrite this property? There is more to adding atoms than just increasing the variable.
Also please note the inconsistency with the "<NATOMS" command in the MDI library calls.

Furthermore, there are likely indexing issues for large systems since all expressions must be 64-bit save and also the MPI calls may need some adjustments. For the time being, It may be better to restrict the use of this functionality to cases where `natoms << 2billion`.

Reply:
Yes, the lines in question are definitely insufficient to correctly change the number of atoms in the system.  These lines are really just placeholders for possible eventual support for the ">NATOMS" command.  Currently, LAMMPS does not register the ">NATOMS" command with MDI, and will raise an error if a driver attempts to use it.  The lines can be safely removed, if desired.

MDI now supports communication of 64-bit ints.  For example:

`ierr = MDI_Recv((char *) &atom->natoms, 1, MDI_INT64_T, mdicomm);`

If the driver is using differently sized integers, MDI automatically makes the conversion.  As long as both the engine and driver use types that are sufficient for the full scope of their possible use cases, there shouldn't be any risk of integer overflows.

Reply:
Ok, thanks. I've updated the code to print an error, if it every encounters this statement and added a comment explaining why.